### ISSUE TITLE
[Bug] Closing and opening some interfaces in quick succession can cause the latter to get stuck on screen without focus

### ISSUE THREAD

**Issue by jag3dagster** [overextended/ox_lib#61]
**Description:**
When attempting to open a new NUI interface option immediately after a NUI interface was closed can cause the newer interface to get stuck on screen without NUI focus. Opening up a third NUI interface give control back and sometimes allows you to clear the NUI menus off screen.

*Note:* I was unable to repro this with two alertDialogs consistently. This may be specific to two separate types of interfaces being closed and opened.

**Expected**
When opening a second NUI interface immediately after closing an initial NUI interface, the user remains in NUI focus and can interact with the second NUI interface.

**Actual**
The second NUI interface get stuck on screen without focus and the user cannot interact with it.

**Repro Steps**
1. Run the following code block
```lua
lib.inputDialog('test', { 'text here' })

lib.alertDialog({
    header = 'stuck',
    content = 'I think I am stuck here.',
    cancel = true
})
```
2. Close out of the inputDialog
3. 'stuck' alertDialog should appear but NUI focus is lost

**Comment by jag3dagster** [comment]
Is there a public roadmap / hope for features?

**Comment by LukeWasTakenn** [comment]
There is not, other than open feature requests on the Issues page.

**Comment by LukeWasTakenn** [comment]
Resolved in commit 953673690c02c05ef133e6676c432d788f9c19f2.

Turns out this was a race condition where the focus for input was being set after resolving the promise instead of before.

Handling nui focus between different elements (like triggering a menu and then a input dialog) is going to be quite difficult since the menu does not need mouse control but the input does, so we still need to get to a solution for that.
